### PR TITLE
Move theming images to AppData

### PR DESCRIPTION
**Pull request by juliusknorr**
Since we have a dedicated app data folder there is no need to store theming files directly in the data folder any more. 

Fixes https://github.com/nextcloud/server/issues/1725

### CODE REVIEW

File: apps/theming/lib/Controller/ThemingController.php [hunk 1]
```diff
@@ -260,45 +266,49 @@ public function undo($setting) {
 	 * @PublicPage
 	 * @NoCSRFRequired
 	 *
-	 * @return StreamResponse|NotFoundResponse
+	 * @return Http\FileDisplayResponse|NotFoundResponse
```
Comment (nickvergessen) [inline apps/theming/lib/Controller/ThemingController.php]: Please fully import the class name

File: apps/theming/tests/Controller/ThemingControllerTest.php [hunk 2]
```diff
@@ -77,6 +79,7 @@ public function setUp() {
 			->method('getTime')
 			->willReturn(123);
 		$this->tempManager = \OC::$server->getTempManager();
+		$this->appData = $this->getMockBuilder('OCP\Files\IAppData')->getMock();
```
Comment (nickvergessen) [inline apps/theming/tests/Controller/ThemingControllerTest.php]: `$this->appData = $this->createMock(\OCP\Files\IAppData::class);`
Same for the other cases

**Review by nickvergessen — CHANGES_REQUESTED** [review]

**Comment by nickvergessen** [comment]
Looks good otherwise

**Comment by juliushaertl** [comment]
@nickvergessen Thanks for the comments. Changes are pushed. :)

File: apps/theming/tests/Controller/ThemingControllerTest.php [hunk 3]
```diff
@@ -58,25 +62,28 @@ class ThemingControllerTest extends TestCase {
...
-		$this->config = $this->getMockBuilder('OCP\IConfig')->getMock();
-		$this->template = $this->getMockBuilder('OCA\Theming\ThemingDefaults')
+		$this->request = $this->getMockBuilder(IRequest::class)->getMock();
+		$this->config = $this->getMockBuilder(IConfig::class)->getMock();
+		$this->template = $this->getMockBuilder(ThemingDefaults::class)
 			->disableOriginalConstructor()->getMock();
-		$this->timeFactory = $this->getMockBuilder('OCP\AppFramework\Utility\ITimeFactory')
+		$this->timeFactory = $this->getMockBuilder(ITimeFactory::class)
 			->disableOriginalConstructor()
 			->getMock();
```
Comment (nickvergessen) [inline apps/theming/tests/Controller/ThemingControllerTest.php]: Also replace all calls to `getMockBuilder()` with calls to `createMock()`
It also automatically calls `->disableOriginalConstructor()->getMock()`

**Comment by juliushaertl** [comment]
@nickvergessen Fixed as well, I've restarted the drone build, since it was only the mysqlmb4 build that looked like it has timed out somehow.

File: apps/theming/lib/Controller/ThemingController.php [hunk 4]
```diff
@@ -183,16 +183,22 @@ public function updateLogo() {
 				Http::STATUS_UNPROCESSABLE_ENTITY
 			);
 		}
+
 		$name = '';
+		try {
+			$folder = $this->appData->getFolder('images');
+		} catch (NotFoundException $e) {
+			$folder = $this->appData->newFolder('images');
```
Comment (LukasReschke) [inline apps/theming/lib/Controller/ThemingController.php]: This line doesn't seem to be covered by a test :)

**Review by LukasReschke — CHANGES_REQUESTED** [review]

File: apps/theming/lib/Controller/ThemingController.php (hunk 4 above)
Comment (juliushaertl) [inline apps/theming/lib/Controller/ThemingController.php]: Those are now covered as well. :)

**Comment by juliushaertl** [comment]
@nickvergessen @LukasReschke Thanks for your comments. Fixes have been pushed. :wink:

**Review by nickvergessen — APPROVED** [review]
Works

**Comment by juliushaertl** [comment]
I've just pushed the last commit again with the missing signoff message.

**Review by LukasReschke — APPROVED** [review]

**Comment by LukasReschke** [comment]
🚀

**Pull request merged**